smdebug/core/hook.py

```diff
@@ -668,6 +673,9 @@ def save_scalar(self, name, value, sm_metric=False, timestamp: float = None):
         :param sm_metric: True/False. If set to True, the scalar value will be written to
         SageMaker
         """
+        if self._is_not_supported():
+            # Do not log scalars if smdebug hook is not supported
```

Comment:
smdebug hook is not supported means? What is the case?

Reply:
If smdebug hook is not supported by the training strategy.

See: https://github.com/awslabs/sagemaker-debugger/blob/e6359444d28ae7de260074ef4d5962ac83fa649a/tests/tensorflow2/test_keras_mirrored.py#L417